plugins/generator-1.16.5/forge-1.16.5/templates/item.java.ftl

```diff
@@ -121,7 +121,7 @@ import net.minecraft.entity.ai.attributes.Attributes;
 			setRegistryName("${registryname}");
 		}
 
-        <#if data.animation != "eat">
+        <#if (data.animation != "eat" && data.isFood) || (data.animation != "none" && !data.isFood)>

```

Comment:
This can likely be simplified.
```suggestion
        <#if data.animation != (data.isFood?then("eat", "none"))>
```
Even better if you do these checks using `Item.hasDefaultAnimation()`.